mesonbuild/dependencies/base.py

```diff
@@ -308,17 +309,37 @@ def is_built(self) -> bool:
     def get_partial_dependency(self, *, compile_args: bool = False,
                                link_args: bool = False, links: bool = False,
                                includes: bool = False, sources: bool = False,
-                               extra_files: bool = False) -> InternalDependency:
-        final_compile_args = self.compile_args.copy() if compile_args else []
-        final_link_args = self.link_args.copy() if link_args else []
-        final_libraries = self.libraries.copy() if links else []
-        final_whole_libraries = self.whole_libraries.copy() if links else []
-        final_sources = self.sources.copy() if sources else []
-        final_extra_files = self.extra_files.copy() if extra_files else []
-        final_includes = self.include_directories.copy() if includes else []
+                               extra_files: bool = False, only_exts: bool = False) -> InternalDependency:
+        final_compile_args = self.compile_args.copy() if compile_args and not only_exts else []
+        final_link_args = self.link_args.copy() if link_args and not only_exts else []
+        final_libraries = self.libraries.copy() if links and not only_exts else []
+        final_whole_libraries = self.whole_libraries.copy() if links and not only_exts else []
+        final_sources = self.sources.copy() if sources and not only_exts else []
+        final_extra_files = self.extra_files.copy() if extra_files and not only_exts else []
+        final_includes = self.include_directories.copy() if includes and not only_exts else []
         final_deps = [d.get_partial_dependency(
             compile_args=compile_args, link_args=link_args, links=links,
-            includes=includes, sources=sources) for d in self.ext_deps]
+            includes=includes, sources=sources, only_exts=False) for d in self.ext_deps]
+        if only_exts:
+            for library in self.libraries:
+                final_deps += [d.get_partial_dependency(
+                    compile_args=compile_args, link_args=link_args, links=links,
+                    includes=includes, sources=sources, only_exts=False) for d in library.external_deps]
```

Comment:
This gets repeated so much I think it would be worth writing a helper functions like:
```python
def make_deps(deps: T.Iterable[Dependency]) -> T.List[Dependency]:
    return [d.get_partial_dependency(...) for d in deps]
```